lib/src/Base/Func/PointToPointEvaluation.cxx

```diff
@@ -0,0 +1,298 @@
+//                                               -*- C++ -*-
+/**
+ * @brief The class that implements the composition between evaluations or 
+ *        field to point and point to field functions
+ *
+ * Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ * Permission to copy, use, modify, sell and distribute this software
+ * is granted provided this copyright notice appears in all copies.
+ * This software is provided "as is" without express or implied
+ * warranty, and with no claim as to its suitability for any purpose.
+ *
+ *
+ */
+
+#include "openturns/PointToPointEvaluation.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/NoEvaluation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(PointToPointEvaluation);
+
+static const Factory<PointToPointEvaluation> Factory_PointToPointEvaluation;
+
+/* Default constructor */
+PointToPointEvaluation::PointToPointEvaluation()
+  : EvaluationImplementation()
+  , isFunctionComposition_(true)
+  , leftFunction_()
+  , rightFunction_()
+  , fieldToPointFunction_()
+  , pointToFieldFunction_()
```

Comment:
warning: field 'fieldToPointFunction_' will be initialized after field 'pointToFieldFunction_' [-Wreorder]

Reply:
The attributes order in the initialization list must be the same as their declaration order in the header file.